### PR TITLE
GDB-9115: Fixes Tab Behavior of dialogs

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
When opening a dialog, there are some unexpected behaviors:

- A visible scrollbar on the left side.
- A line on the left side that is not overlaid by the dialog overlay.
- Tab functionality not working.

## Why
- The page has overflow auto.
- Setting the width to 100% on the dialog-overlay somehow broke the calculation of the overlay width.
- Not implemented.

## How
- Remove the overflow property of the body when the dialog is open and return it when closed.
- The width property has been removed.
- Implemented functionality that controls active elements to be only those belonging to the dialog.

### CODE REVIEW

File: ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.tsx [hunk 1]
```diff
@@ -12,16 +15,75 @@ export type DialogConfig = {
...
+   * @param ev The keyboard event.
+   */
+  @Listen('keydown', {target: "document"})
+  keydownListener(ev: KeyboardEvent) {
+    if (ev.key === 'Escape') {
+      this.closeButton.click();
+    }
+
+    // Handle Tab key press to manage focus within the dialog
+    if (ev.key === 'Tab') {
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.tsx:40]: This approach is far more complicated that it should be IMO. Can't we just constrain the focus in the dialog preventing it from popping-out to elements outside? https://stackoverflow.com/a/51210194 check this approach

File: ontotext-yasgui-web-component/src/services/utils/dialog-util.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {HtmlUtil} from './html-util';
+
+export class DialogUtil {
+  static  preventLeavingDialog(hostElement: HTMLElement, ev: KeyboardEvent) {
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/utils/dialog-util.ts:4]: Move this in the HtmlUtil. It has nothing particularly specific related with dialogs.

Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/services/utils/dialog-util.ts]: Done

File: ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.tsx (hunk 1 above)
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.tsx]: Done

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**